### DISCUSSION TITLE
Handling expo Deep Links with EAS Builds

### DISCUSSION THREAD

**Discussion by sumitsharma7667** [expo/expo#30288] [Archived: Questions]
I’m currently working on integrating deep linking into my Expo app using EAS (Expo Application Services) builds, and I could use some guidance on the best practices for setup and configuration. Here’s what I’ve done so far and where I need assistance:

1. Configuration in app.json:
I have defined a custom URL scheme under the expo configuration in my app.json file using the following:
`{
  "expo": {
    "scheme": "rentcreek"
  }
}`
2. Handling Deep Links:
I’m using expo-linking to handle incoming deep links within my app. Here’s a basic implementation:

`
import React, { useContext } from 'react';
import { Text } from 'react-native';
import * as Linking from 'expo-linking';
import { NavigationContainer } from '@react-navigation/native';
import BottomNavigation from '../components/navigation/bottomNavigation';
import { useUser } from '../lib/context/userContext';

import RootscreenStack from '../components/navigation/rootStack';
export default function Navigation() {

  const { user } = useUser();
  const prefix = Linking.createURL('/');
  
const config = {
    screens: {
      MagicLinkSignup: {
        path: "MagicLinkSignup/:token",
        parse: (token: String) => `${token}`
      },
      AcceptLeaseByLink: {
        path: "AcceptLeaseByLink/:token",
        parse: (token: String) => `${token}`
      },
      Splash: "Splash"

    }
  }
  const linking = {
    prefixes: [prefix, 'https://', 'rentcreek://'],
    config
  };
  return (
    <NavigationContainer linking={linking} fallback={<Text>Loading...</Text>}>
      {user?.role == "landlord" || user?.role == "admin" || user?.role == "superadmin" ?
        <BottomNavigation />
        :
        <RootscreenStack />
      }
    </NavigationContainer>
  )}
`

3. Testing and Deployment:

During development, I test deep links by manually entering them (rentcreek://MagicLinkSignup/dfdg) in the device's browser or sending them via email.
For production deployment with EAS, I plan to use eas build command.
4. Questions:

Do I need to configure anything additional for production deployment regarding deep links?
What are the best practices or common pitfalls to watch out for when handling deep links with EAS builds?
I would appreciate any advice, tips, or resources you can share to help me ensure a smooth integration of deep linking in my Expo app using EAS builds.

Thank you in advance for your support!

**Comment by atultiwaree** [comment]
Any update on this ?
